### DISCUSSION TITLE
Propagation of a bond between two polymers

### DISCUSSION THREAD

**Discussion by nmendiboure** [glotzerlab/hoomd-blue#1738] [General]
Hello everyone,

I'm new to HOOMD-blue and to molecular dynamics as a whole. For a PhD side project, I need to simulate the attachment of homologous chromosomes during meiosis (specifically, the formation of the synaptonemal complex). In my model, chromosomes are represented as bead-spring chains, with their telomeres (the ends of the chains) tethered to the surface of the nuclear envelope via manifold.rattle on a spherical surface. I plan to select some beads at random to act as pioneer contacts. That is, when one of these beads comes within a certain distance of its homologous counterpart on the other chromosome, a bond can be formed between the chromosomes with a certain probability (konkon​). Next, I'd like to model a 'zippering' process where, once a pioneer bond is formed, it can spread to neighboring beads, further linking the chromosomes.

This scenario might seem quite specific, but I'm wondering if there's a built-in feature in HOOMD-blue that could facilitate this, something akin to a 'bond_reaction' capability, perhaps?

If not, should I approach this by manually intervening in the simulation? For example, by taking snapshots at defined intervals, checking for the conditions within the snapshot, modifying the topology as needed, and then reintegrating the modified snapshot back into the simulation to continue it?

Thank you in advance for any guidance you can provide.

**Comment by chrisjonesBSU** [comment]
There is not a built-in bond-forming or reaction mechanism in Hoomd, so it would have to be done manually, but something like this is a perfect use case for the [custom actions](https://hoomd-blue.readthedocs.io/en/v4.5.0/tutorial/04-Custom-Actions-In-Python/00-index.html) functionality where you would basically do what you described.

**Reply by nmendiboure** [reply]
Thank you for the guidance; I'll give that approach a try.
